readme.md

```diff
@@ -118,7 +118,12 @@ Ensure you use this correctly. For example, files and directories should be matc
 
 Type: `string | string[]`
 
-Use `*` to match zero or more characters. A pattern starting with `!` will be negated.
+Use `*` to match zero or more characters.
+
+A leading `!` negates the pattern.
```

Comment:
Readme and index.d.ts should be in sync documentation-wise.

Reply:
"This part feels moot in the TS definition (but not the readme) as it's already made clear by the types."

Did I get you wrong or have you changed your mind?

Reply:
Ok, let me rephrase: index.d.ts and readme should be in sync, except for cases where the readme just points out the types as that's already clear by the types.

However, that's not what I'm pointing out here. You have removed some stuff from index.d.ts that is preserved here and not updated other parts, like https://github.com/sindresorhus/matcher/pull/30/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R83

Reply:
Ok, I restored the missing stuff and made the contents as coherent as I could via ee30306.
I also found: 
   - 'inputs' and 'patterns' were randomly used in both plural and singular form. Fixed that.
   - '@param options' is missing in index.d.ts - I left it for you to decide.

Reply:
Ok, I restored the missing stuff and made the contents as coherent as coherent as I could via ee30306.
Also found: 
   - 'inputs' and 'patterns' were randomly used in both plural and singular form. Fixed that.
   - '@param options' is missing in index.d.ts - I left it for you to decide.